### PR TITLE
fix(generic-sync): Added check for status subresource on resource import

### LINKED ISSUE #1499: Object status is not synced from host objects to virtual objects
### What happened?

I have enabled this `sync.generic.config` in a vcluster in multi namespace mode

```
sync:
  generic:
    role:
      extraRules:
        - apiGroups: ["cert-manager.io"]
          resources: ["issuers", "certificates", "certificaterequests"]
          verbs: ["create", "delete", "patch", "update", "get", "list", "watch"]
        - apiGroups: ["cert-manager.io"]
          resources: ["clusterissuers"]
          verbs: ["get", "list", "watch"]
        - apiGroups: ["acme.cert-manager.io"]
          resources: ["orders"]
          verbs: ["create", "delete", "patch", "update", "get", "list", "watch"]
    clusterRole:
      extraRules:
        - apiGroups: ["apiextensions.k8s.io"]
          resources: ["customresourcedefinitions"]
          verbs: ["get", "list", "watch"]
    config: |-
      version: v1beta1
      export:
        - apiVersion: cert-manager.io/v1
          kind: Certificate
        - apiVersion: cert-manager.io/v1
          kind: Issuer
      import:
        - kind: Secret
          apiVersion: v1
        - apiVersion: cert-manager.io/v1
          kind: ClusterIssuer
        - apiVersion: acme.cert-manager.io/v1
          kind: Order
        - apiVersion: cert-manager.io/v1
          kind: Certificate
```

Certificate objects created on the host is synced correctly to the virtual cluster, except for the status part of the object. The status is just not available in the imported certificate object.

This is the host namespace, notice the `vcluster.loft.sh/object-name: pktest` label
```
❯ k describe ns vcluster-d9b151e9-2f96135e
Name:         vcluster-d9b151e9-2f96135e
Labels:       kubernetes.io/metadata.name=vcluster-d9b151e9-2f96135e
              pod-security.kubernetes.io/enforce=baseline
              pod-security.kubernetes.io/warn=restricted
              vcluster.loft.sh/label-vcluster-vk8s1-x-vk8s1-x-cf1227b7b2=pktest
              vcluster.loft.sh/managed-by=vcluster-vk8s1-x-vk8s1
              vcluster.loft.sh/vcluster-name=vk8s1
              vcluster.loft.sh/vcluster-namespace=vcluster-vk8s1
Annotations:  policies.kyverno.io/last-applied-patches:
                add-baseline-enforce-restricted-warn.add-psa-labels.kyverno.io: added /metadata/labels/pod-security.kubernetes.io~1warn
              vcluster.loft.sh/object-name: pktest
              vcluster.loft.sh/object-uid: 26209e71-249c-4220-ba1f-eb6bb90e1da6
Status:       Active
```

Into this namespace on the host, I deployed a `testcert2` certificate object. The status present.
```
❯ k get cert -n vcluster-d9b151e9-2f96135e
NAME        READY   SECRET      AGE
testcert2   True    testcert2   2m52s

❯ k describe cert -n vcluster-d9b151e9-2f96135e
Name:         testcert2
Namespace:    vcluster-d9b151e9-2f96135e
Labels:       <none>
Annotations:  vcluster.loft.sh/controlled-by: certificate/cert-manager.io/v1/GenericImport
API Version:  cert-manager.io/v1
Kind:         Certificate
Metadata:
  Creation Timestamp:  2024-01-31T20:47:37Z
  Generation:          1
  Resource Version:    83401906
  UID:                 06dce37b-8c8f-4166-bae8-e634c337dfa8
Spec:
  Dns Names:
    testcert2.vk8s1.test.vcluster.geodata.ek8s1.geopoint.no
  Issuer Ref:
    Group:      cert-manager.io
    Kind:       ClusterIssuer
    Name:       letsencrypt-staging
  Secret Name:  testcert2
Status:
  Conditions:
    Last Transition Time:  2024-01-31T20:47:44Z
    Message:               Certificate is up to date and has not expired
    Observed Generation:   1
    Reason:                Ready
    Status:                True
    Type:                  Ready
  Not After:               2024-04-30T19:47:39Z
  Not Before:              2024-01-31T19:47:40Z
  Renewal Time:            2024-03-31T19:47:39Z
  Revision:                1
Events:
  Type    Reason     Age    From                                       Message
  ----    ------     ----   ----                                       -------
  Normal  Issuing    3m29s  cert-manager-certificates-trigger          Issuing certificate as Secret does not exist
  Normal  Generated  3m28s  cert-manager-certificates-key-manager      Stored new private key in temporary Secret resource "testcert2-2jqcj"
  Normal  Requested  3m28s  cert-manager-certificates-request-manager  Created new CertificateRequest resource "testcert2-grrn4"
  Normal  Issuing    3m22s  cert-manager-certificates-issuing          The certificate has been successfully issued
```

The testcert2 object is correctly imported to the pktest namespace in the vcluster. But the status is not present.
```
❯ k get cert -n pktest
NAME        READY   SECRET      AGE
testcert2           testcert2   5m25s

❯ k describe cert -n pktest
Name:         testcert2
Namespace:    pktest
Labels:       <none>
Annotations:  vcluster.loft.sh/controlled-by: certificate/cert-manager.io/v1/GenericImport
API Version:  cert-manager.io/v1
Kind:         Certificate
Metadata:
  Creation Timestamp:  2024-01-31T20:47:38Z
  Generation:          1
  Resource Version:    1916
  UID:                 9a17da83-21a6-440d-91cc-2925838ac145
Spec:
  Dns Names:
    testcert2.vk8s1.test.vcluster.geodata.ek8s1.geopoint.no
  Issuer Ref:
    Group:      cert-manager.io
    Kind:       ClusterIssuer
    Name:       letsencrypt-staging
  Secret Name:  testcert2
Events:         <none>
```

### What did you expect to happen?

The status of the certificate object should be synced from the host to the virtual cluster.

### How can we reproduce it (as minimally and precisely as possible)?

Create a certificate object in a vcluster namespace on the host.

### Anything else we need to know?

_No response_

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
v1.28.4-eks-8cb36c9
```

</details>

### Host cluster Kubernetes distribution

<details>

```
EKS
```

</details>

### vlcuster version

<details>

```console
$ vcluster --version
0.19.0-alpha.3. I have observed similar behaviour on version 0.18.1
```

</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
k3s: v1.28.5+k3s1
```

</details>

### OS and Arch

<details>

```
OS: Amazon Linux 2
Arch: AMD64
```
</details>

### PR DESCRIPTION
**Pull request by ThomasK33**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #1499

**Please provide a short message that should be published in the vcluster release notes**
Fixed an issue where the vCluster generic sync would not detect whether a CRD has a status subresource on import.

**What else do we need to know?**

### CODE REVIEW

File: pkg/util/translate/translate.go [hunk 1]
```diff
@@ -325,32 +351,30 @@ func ConvertKindToResource(config *rest.Config, groupVersionKind schema.GroupVer
 	return schema.GroupVersionResource{}, kerrors.NewNotFound(schema.GroupResource{Group: groupVersionKind.Group}, groupVersionKind.Kind)
 }
 
-// KindExistsAndIsClusterScoped checks if given CRDs exist in the given group.
+var ErrKindNotFound = errors.New("kind not found")
+
+// KindExists checks if given CRDs exist in the given group.
 // Returns foundKinds, notFoundKinds, error
```
Comment (pascalbreuninger) [inline pkg/util/translate/translate.go:357]: nit: comment doesn't match return struct

**Review by FabianKramm — APPROVED** [review]
LGTM!

File: pkg/util/translate/translate.go (hunk 1 above)
Comment (FabianKramm) [reply pkg/util/translate/translate.go:357]: Yeah please change that then we can merge it

**Review by pascalbreuninger — APPROVED** [review]

File: pkg/util/translate/translate.go (hunk 1 above)
Comment (ThomasK33) [reply pkg/util/translate/translate.go]: Good catch. Fixed. Thanks.

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**